MekHQ/src/mekhq/campaign/personnel/AwardsFactory.java

```diff
@@ -143,30 +145,36 @@ private void loadAwards(){
         }
         
         for(File file : files) {
+            try{
+                InputStream inputStream = new FileInputStream(file);
+                loadAwardsFromStream(inputStream, file.getName());
+            }
+            catch (FileNotFoundException e){
+                MekHQ.getLogger().error(AwardsFactory.class, "loadAwards", e);
+            }
+        }
+    }
 
-            AwardSet awardSet = null;
+    public void loadAwardsFromStream(InputStream inputStream, String fileName){
+        AwardSet awardSet = null;
 
-            try {
-                InputStream inputStream = new FileInputStream(file);
-                JAXBContext jaxbContext = JAXBContext.newInstance(AwardSet.class, Award.class);
-                Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(AwardSet.class, Award.class);
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
 
-                awardSet = unmarshaller.unmarshal(MekHqXmlUtil.createSafeXmlSource(inputStream), AwardSet.class).getValue();
+            awardSet = unmarshaller.unmarshal(MekHqXmlUtil.createSafeXmlSource(inputStream), AwardSet.class).getValue();
 
-                Map<String, Award> tempAwardMap = new HashMap<>();
-                String currentSetName = file.getName().replaceFirst("[.][^.]+$", "");
-                for (Award award : awardSet.getAwards()){
-                    award.setSet(currentSetName);
-                    tempAwardMap.put(award.getName(), award);
-                }
-                awardsMap.put(currentSetName, tempAwardMap);
-            } catch (SAXException | ParserConfigurationException e) {
-                MekHQ.getLogger().error(AwardsFactory.class, "loadAwards", e);
-            } catch (FileNotFoundException e) {
-                MekHQ.getLogger().error(AwardsFactory.class, "loadAwards", "Cannot find XML awards file: " + file.getName(), e);
-            } catch (JAXBException | IOException e) {
-                MekHQ.getLogger().error(AwardsFactory.class, "loadAwards", "Error loading XML for awards", e);
+            Map<String, Award> tempAwardMap = new HashMap<>();
+            String currentSetName = fileName.replaceFirst("[.][^.]+$", "");
```

Comment:
Are you trying to strip the extension from the file name?

Reply:
Yessir.